pkg/agent/billing.go

```diff
@@ -52,14 +53,14 @@ func (m *metricsTimeSlice) Duration() time.Duration { return m.endTime.Sub(m.sta
 
 type vmMetricsInstant struct {
 	// cpu stores the cpu allocation at a particular instant.
-	cpu uint16
+	cpu api.MilliCPU
```

Comment:
Thinking about this more, I think `cpu` would actually be better off as a float all the way through. The difference should be negligible though, so happy to leave that for a follow-up.